caffe-distri/src/main/cpp/util/rdma_sync.cpp

```diff
@@ -89,13 +96,20 @@ void RDMASync<Dtype>::CreateWorkerBuffers(int peer) {
   // Send diff from local (rank_) to remote (peer)
   uint8_t* diff = reinterpret_cast<uint8_t*>(diff_ + offs);
   diff_send_[peer].reset(new RDMABuffer(channel, diff, size));
+
+  // Recv ctrl from remote (peer) to local (rank_)
+  uint8_t* ctrl;
+  CUDA_CHECK(cudaMalloc(&ctrl, CTRL_BUF_SIZE));
```

Comment:
It seems you are doing 1 byte RDMA WRITE IMM DATA for control sync.   I guess you do not dependent on that 1 byte content but depending only on recv event imm_data.  right? If that is the case, may be you can use malloc() memory here.